include/deal.II/lac/solver_control.h

```diff
@@ -528,6 +528,74 @@ class IterationNumberControl : public SolverControl
                        const double   check_value);
 };
 
+
+/**
+ * Specialization of @p SolverControl which returns SolverControl::State::success if
+ * and only if a certain positive number of consecutive iterations satisfy the
+ * specified tolerance. This is useful in cases when solving nonlinear problems
+ * using inexact Hessian.
+ *
+ * For example: The requested number of consecutively converged iterations is 2,
+ * the tolerance is 0.2. The ConsecutiveControl will return SolverControl::State::success
+ * only at the last step in the sequence 0.5, 0.0005, 1.0, 0.05, 0.01.
+ *
+ * @author Denis Davydov, 2017
+ */
+class ConsecutiveControl : public SolverControl
```

Comment:
Though I wished that we had started to name all of these solver control classes with a common *prefix* rather than suffix. That would have made understanding their names so much simpler. But that boat has sailed.